test: add code snippets for using bigframes.ml

ashleyxuu opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

ashleyxuu committed:
test: add code snippets for using bigframes.ml

ashleyxuu commented:
The code snippets here are prepared for the content of "Use BigQuery DataFrames" page.
Content: https://docs.google.com/document/d/1N5mbfoTh63SwTRFkqs-j9m_2WIykDKHPKAECvKScDSg/edit#heading=h.nq1i22bqeo4y

tswast commented on samples/snippets/clustering_model_test.py:
```diff
@@ -0,0 +1,34 @@
+    bq_df = bpd.read_gbq(query_or_table)
+    # [START bigquery_dataframes_clustering_model]
+    # Create the KMeans model
+    cluster_model = KMeans(n_clusters=10)
```
This sample isn't really "standalone". Needs imports at a minimum.

tswast commented on samples/snippets/regression_model_test.py:
```diff
@@ -0,0 +1,53 @@
+def test_regression_model():
+    import bigframes.pandas as bpd
+
+    query_or_table = "bigquery-public-data.ml_datasets.penguins"
```
This is a lot of setup. Might be best to combine with samples/snippets/clean_and_prep_ml_data_test.py

These aren't really "standalone" samples, as they require so much setup that we aren't showing the user.

ashleyxuu replied:
Done making it a single region and deleting "clean_and_prep_ml_data_test"

tswast reviewed (commented):
My main concern with merging this PR is that running some of these samples may take a long time. It might be best to create a separate folder for these and exclude them from our presubmit samples tests and move the ML samples tests to the e2e job.

ashleyxuu committed:
test: add code snippets for loading data from BigQuery Job (#154)

* test: add code snippets for loading data from BigQuery Job

* fix: address the comments

* fix: fix the broken test

* use BigQuery Client library to get the job_id

* feat: Implement operator `@` for `DataFrame.dot` (#139)

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://togithub.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes b/297502513 🦕

* fix: fix the comments

---------

Co-authored-by: Shobhit Singh <shobs@google.com>

ashleyxuu committed:
move ml samples to e2e, make the tests standalone

orrbradford committed:
feat: add bigframes.options.compute.maximum_bytes_billed option that sets maximum bytes billed on query jobs (#133)

-implement context manager for global options
-maximum_bytes_billed only applies to query jobs. This limitation will be set per query. Operations that trigger multiple jobs may result in total usage beyond this setting

tswast committed:
docs: fix indentation on `read_gbq_function` code sample (#163)

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://togithub.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

ashleyxuu committed:
Merge branch 'main' into add-code-snippets-use-ml

milkshakeiii committed:
feat: add pd.get_dummies (#149)

* feat: add pd.get_dummies

* remove unneeded prefix case

* param/documentation fixes

* be stricter about types in test

* be stricter about types in series test

* remove unneeded comment

* adjust for type difference in pandas 1

* add example code (tested)

* fix None columns and add test cases

* variable names and _get_unique_values per-column

* account for pandas 1 behavior difference

* remove already_seen set

* avoid unnecessary join/projection

* fix column ordering edge case

* adjust for picky examples checker

* example tweak

* make part of the example comments

* use ellipsis in doctest comment

* add <BLANKLINES> to doctest string

* extract parameter standardization

* extract submethods

---------

Co-authored-by: Henry J Solberg <henryjsolberg@google.com>

ashleyxuu committed:
fix: fix the failed test

ashleyxuu committed:
Merge branch 'main' into add-code-snippets-use-ml

ashleyxuu committed:
fix: reorganize the directory

ashleyxuu committed:
Merge remote-tracking branch 'origin/add-code-snippets-use-ml' into add-code-snippets-use-ml

ashleyxuu committed:
Merge branch 'main' into add-code-snippets-use-ml

tswast reviewed (approved):
Thanks! These are great.
